projects/ngrx.io/content/guide/component-store/comparison.md

```diff
@@ -0,0 +1,100 @@
+
+<div class="alert is-important">
+
+**Note:** It's important to understand what the cost is and why we are adding it.
+
+</div>
+
+Both of them bring [push-based architecture](https://medium.com/@thomasburlesonIA/push-based-architectures-with-rxjs-81b327d7c32d), which is the first indirection. User can no longer 
+get the result of a service method call, instead they would be listening for Observable values
+exposed by that service. The benefit, on the other side, is that User no longer has to worry what
+is changing the state - all the component needs to know is that something has changed it. If the
+component wants to change the state itself, it sends the message about it (either dispatches an 
+Action in Store, or calls ComponentStore's updater or effect).
+
+Actions are the second indirection. They are present in Global Store only. There are many benefits 
+of this indirection, such as:
+* ability to trigger multiple effects/reducers at the same time
+* greater scalability
+* helpful DevTools
+
+ComponentStore doesn't have that indirection, however it also loses the above mentioned benefits.
+
+The scale of state that it works with has to be smaller, with brings another set of benefits, such as:
+* ComponentStore that is tied to the specific node in the components tree, will be automatically cleaned up when that node is destroyed
+* state is fully self-contained with each ComponentStore, and thus allows to have multiple 
+independent instances of the same component
+* provides simpler state management to small/medium sized apps
+
+<div class="alert is-helpful">
+
+The difference between the benefits and trade-offs of Stores make Global Store better suited for
+managing global shared state, where ComponentStore shines managing more local, encapsulated state,
+as well as component UI state.
+
+</div>
+
+## State ownership
+
+Store works with the **single** immutable object, that contains all of the shared state throughout
+the application. There are multiple reducers, each one responsible for a particular **slice** of
+state.
+
+Each ComponentStore is fully responsible for a its own state. There could be **many** different
+ComponentStores, but each one should store their own distinct state.
+
+<figure>
+  <img src="generated/images/guide/component-store/state-structure.png" alt="Comparison of NgRx Store and Component Store state ownership or placement" width="100%" height="100%" />
+</figure>
+
+## File structure
+
+ComponentStore is focused on smaller part of state, and thus should contain not only the state
+itself, but also every "prescription" of how it could be changed. All "`updater`s" and "`effect`s"
+should be part of the ComponentStore, responsible for the specific state.
+
+It makes ComponentStore less scalable - if there are too many updaters and effects in a single class,
+then it quickly becomes unreadable.
+
+Shared `select`ors should also be part of the ComponentStore, however downstream components might
```

Comment:
I'm confused with `ComponentStore<string>`. Should it just say that it's acceptable to create a new ComponentStore for that component?

Reply:
It should've been `ComponentStore<object>` 🤦‍♂️